cpp/src/orderby/orderby.cu

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2018-2019 BlazingDB, Inc.
+ *     Copyright 2018 Jean Pierre Huaroto <jeanpierre@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <type_traits>
+
+#include "cudf.h"
+#include "utilities/cudf_utils.h"
+#include "utilities/error_utils.h"
+
+#include "rmm/thrust_rmm_allocator.h"
+
+#include "../sqls/sqls_rtti_comp.h"
+
+namespace{ //annonymus
+
+  // thrust::device_vector set to use rmmAlloc and rmmFree.
+  template<typename T>
+  using Device_Vector = thrust::device_vector<T, rmm_allocator<T>>;
+
+  gdf_error multi_col_order_by(gdf_column** cols,
+                               size_t ncols,
+                               char* asc_desc,
+                               gdf_column* output_indices,
+                               bool flag_nulls_are_smallest)
+  {
+    GDF_REQUIRE(cols != nullptr && output_indices != nullptr, GDF_DATASET_EMPTY);
+    GDF_REQUIRE(cols[0]->size == output_indices->size, GDF_COLUMN_SIZE_MISMATCH);
+
+    // Check for null so we can use a faster sorting comparator 
+    bool have_nulls = false;
+    for (size_t i = 0; i < ncols; i++) {
+      if (cols[i]->null_count > 0) {
+        have_nulls = true;
+        break;
+      }
+    }
+
+    Device_Vector<void*> d_cols(ncols);
+    Device_Vector<gdf_valid_type*> d_valids(ncols);
+    Device_Vector<int> d_types(ncols, 0);
+
+    void** d_col_data = d_cols.data().get();
+    gdf_valid_type** d_valids_data = d_valids.data().get();
+    int* d_col_types = d_types.data().get();
+
+    soa_col_info(cols, ncols, d_col_data, d_valids_data, d_col_types);
+
+    switch(output_indices->dtype)
```

Comment:
Remove the `switch` in favor of a `GDF_REQUIRE( output_indices->dtype == GDF_INT32, GDF_UNSUPPORTED_DTYPE)` with a comment that explains why only `int32_t` is supported.